### ISSUE TITLE
Neo.DatabaseError.Statement.ExecutionFailed because of different query plans

### ISSUE THREAD

**Issue by mmopitz** [neo4j/neo4j#12862]
I get a `Neo.DatabaseError.Statement.ExecutionFailed` when I try to fire off a query.

Problematic Query:
```
match (n:Product)
where n:Product OR n:Category
return n
```
Note: I know that the query could be simplified but I guess this could be a bug in neo4j

## System

**Neo4j Version:** 4.4.5 
**Operating System:** Ubuntu 20.04
**API:** Neo4j-Browser

## Steps to reproduce

I had to use the northwind-example to reproduce the issue because I wasn't able to reproduce the issue with a minimal example.

1. start neo4j `docker run --rm -p 7687:7687 -p 7474:7474 neo4j:4.4.5`
2. connect to the neo4j browser: http://localhost:7474
3. populate the database with a part of the northwind example.
```cypher
LOAD CSV WITH HEADERS FROM "http://data.neo4j.com/northwind/products.csv" AS row
CREATE (n:Product)
SET n = row,
n.unitPrice = toFloat(row.unitPrice),
n.unitsInStock = toInteger(row.unitsInStock), n.unitsOnOrder = toInteger(row.unitsOnOrder),
n.reorderLevel = toInteger(row.reorderLevel), n.discontinued = (row.discontinued <> "0")

LOAD CSV WITH HEADERS FROM "http://data.neo4j.com/northwind/categories.csv" AS row
CREATE (n:Category)
SET n = row

LOAD CSV WITH HEADERS FROM "http://data.neo4j.com/northwind/suppliers.csv" AS row
CREATE (n:Supplier)
SET n = row
```
4. execute the query
```
match (n:Product)
where n:Product OR n:Category
return n
```

### Expected behavior
Query should return the products (since Category and Product are disjoint)

### Actual behavior
I get following exception:

```
2022-04-11 14:53:58.111+0000 ERROR Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: 
Differences:
 - QueryGraph
    A: QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Product OR n:Category']}
    B: QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Category', 'n:Product OR n:Category']}
, reference 90f317e2-f325-481f-a3c9-bdd0021ff2dc, queryId: 404.
Expected 
RegularSinglePlannerQuery(QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Product OR n:Category']},InterestingOrder(RequiredOrderCandidate(List()),List()),RegularQueryProjection(Map(n -> Variable(n)),QueryPagination(None,None),Selections(Set())),None,None) 

Instead, got: 
RegularSinglePlannerQuery(QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Category', 'n:Product OR n:Category']},InterestingOrder(RequiredOrderCandidate(List()),List()),RegularQueryProjection(Map(n -> Variable(n)),QueryPagination(None,None),Selections(Set())),None,None)
Plan: ProduceResult(List(n)) {
  LHS -> Selection(Ands(Vector(Ors(List(HasLabels(Variable(n),List(LabelName(Product))), HasLabels(Variable(n),List(LabelName(Category)))))))) {
    LHS -> Distinct(Map(n -> Variable(n))) {
      LHS -> Union() {
        LHS -> NodeByLabelScan(n, LabelName(Product), Set(), IndexOrderAscending) {}
        RHS -> Selection(Ands(Vector(HasLabels(Variable(n),List(LabelName(Product)))))) {
          LHS -> NodeByLabelScan(n, LabelName(Category), Set(), IndexOrderAscending) {}
        }
      }
    }
  }
} 


Differences:
 - QueryGraph
    A: QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Product OR n:Category']}
    B: QueryGraph {Nodes: ['n'], Predicates: ['n:Product', 'n:Category', 'n:Product OR n:Category']}
, reference 714a873c-508c-488c-bca3-bc0dd9f2d14c, queryId: 501.
```

**Comment by mnd999** [comment]
Thanks for the report, that looks like a bug. Wie'll look into it and let you know the fix.

**Comment by arnefischereit** [comment]
Hi @mmopitz,

This is caused by the same problem as #12818 and a fix will be released with the upcoming 4.4.6 release.
